File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s1: str, s2: str) -> bool:
        d = {}
        ana = False
        if len(s1)!=len(s2):
            return False
        for i in s1:
            if i in d:
                d[i]+=1
            else:
                d[i] = 1
        ana = True
        for i in s2:
            if i in d:
                d[i]-=1
            else:
                ana = False
        if ana:
            for i in d.values():
                if i !=0:
                    ana = False
                else:
                    ana = True
        return ana

File: 0242-valid-anagram/0242-valid-anagram.java
import java.util.*;

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
            return false;
        int[] chs = new int[26];
        int[] cht = new int[26];
        // System.out.println(Arrays.toString(chs));
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            char ch1 = t.charAt(i);
            chs[ch - 'a']++;
            cht[ch1-'a']++;
        }
        System.out.println(Arrays.toString(chs));
        for (int i = 0; i < chs.length; i++) {
            if(chs[i]!=cht[i]) return false;
        }
        return true;
    }
}